universal-application-tool-0.0.1/app/assets/javascripts/azure_upload.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * This is responsible for uploading a file to Azure blob storage.
+ */
+class AzureUploadController {
+  static UPLOAD_CONTAINER_ID = 'azure-upload-form-component';
+
+  constructor() {
+    const uploadContainer = document.getElementById(AzureUploadController.UPLOAD_CONTAINER_ID);
+    uploadContainer.addEventListener("submit", (event) => this.attemptUpload(event, uploadContainer));
+  }
+
+  getValueFromInputLabel(label: string): string {
+    return (<HTMLInputElement>document.getElementsByName(label)[0]).value;
+  }
+
+  attemptUpload(event: Event, uploadContainer: HTMLElement | null) {
+    event.preventDefault();
+    const azblob = window["azblob"];

```

Comment:
question (blocking): what sets this value? why not `window.azblob`?

Reply:
So this is set by the Azure javascript client that we're importing by setting a script tag that links to a CDN. And I tried `window.azblob` which worked with regular Javascript, but the typescript compiler didn't like it.

Reply:
Gotcha. Please add a comment explaining that